plugins/outputs/otlp/otlp.go

```diff
@@ -0,0 +1,379 @@
+		Value:        value,
+	}
+	return &otlpmetricpb.DoubleGauge{
+		DataPoints: []*otlpmetricpb.DoubleDataPoint{double},
+	}
+}
+
+// Write the metrics to OTLP destination
+func (o *OTLP) Write(metrics []telegraf.Metric) error {
+	batch := sorted(metrics)
+	samples := []*otlpmetricpb.ResourceMetrics{}
+	currentTs := time.Now().UnixNano()
+	for _, m := range batch {
+		for _, f := range m.FieldList() {
+			sample, point := o.protoTimeseries(m, f)
+
+			labels := protoStringLabels(m.TagList())
+			ts := m.Time().UnixNano()
+
+			switch m.Type() {
+			case telegraf.Counter:
+				switch v := f.Value.(type) {
+				case uint64:
+					if v <= uint64(MaxInt) {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, int64(v)),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, int64(MaxInt)),
+						}
+					}
+				case int64:
+					point.Data = &otlpmetricpb.Metric_IntSum{
+						IntSum: monotonicIntegerPoint(labels, ts, currentTs, v),
+					}
+				case float64:
+					point.Data = &otlpmetricpb.Metric_DoubleSum{
+						DoubleSum: monotonicDoublePoint(labels, ts, currentTs, v),
+					}
+				case bool:
+					if v {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, 1),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, 0),
+						}
+					}
+				case string:
+					o.Log.Error("get type failed: unsupported telegraf value type string")
+					continue
+				default:
+					o.Log.Errorf("get type failed: unsupported telegraf value type %v\n", f.Value)
+					continue
+				}
+			case telegraf.Gauge, telegraf.Untyped:
+				switch v := f.Value.(type) {
+				case uint64:
+					if v <= uint64(MaxInt) {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, int64(v)),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, int64(MaxInt)),
+						}
+					}
+				case int64:
+					point.Data = &otlpmetricpb.Metric_IntGauge{
+						IntGauge: intGauge(labels, ts, v),
+					}
+				case float64:
+					point.Data = &otlpmetricpb.Metric_DoubleGauge{
+						DoubleGauge: doubleGauge(labels, ts, v),
+					}
+				case bool:
+					if v {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, 1),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, 0),
+						}
+					}
+				case string:
+					o.Log.Error("get type failed: unsupported telegraf value type string")
+					continue
+				default:
+					o.Log.Errorf("get type failed: unsupported telegraf value type %v\n", f.Value)
+					continue
+				}
+			// TODO: add support for histogram & summary
+			case telegraf.Histogram, telegraf.Summary:
+				fallthrough
```

Comment:
Please either implement it or leave it out!

Reply:
Removed